manifests/config.pp

```diff
@@ -2,15 +2,24 @@
 class ntp::config inherits ntp {
 
   if $keys_enable {
-    $directory = ntp_dirname($keys_file)
-    file { $directory:
-      ensure  => directory,
-      owner   => 0,
-      group   => 0,
-      mode    => '0755',
+    file { $keys_file:
+      ensure => file,
+      owner  => 0,
+      group  => ntp,
+      mode   => '0640',
     }
   }
 
+  if $config_dir {
+      file { $config_dir:
+        ensure  => directory,
+        owner   => 0,
+        group   => ntp,
```

Comment:
No data about this in the metadata. Either we have add the metadata (and check it for all systems, which I can't, since I don't have access to an AIX machine) or I modify this part back to 0 (which I have, since the previous option is impossible for me to achieve)